pkg/apis/serving/v1alpha1/configuration_validation_test.go

```diff
@@ -71,6 +72,86 @@ func TestConfigurationSpecValidation(t *testing.T) {
 			},
 		},
 		want: apis.ErrDisallowedFields("revisionTemplate.spec.container.name"),
+	}, {
+		name: "build is a BuildSpec",
+		c: &ConfigurationSpec{
+			Build: &RawExtension{
+				BuildSpec: &buildv1alpha1.BuildSpec{
+					Steps: []corev1.Container{{
+						Image: "foo",
+					}},
+				},
+			},
+			RevisionTemplate: RevisionTemplateSpec{
+				Spec: RevisionSpec{
+					Container: corev1.Container{
+						Image: "hellworld",
+					},
+				},
+			},
+		},
+		want: nil,
+	}, {
+		name: "build is an Object",
+		c: &ConfigurationSpec{
+			Build: &RawExtension{
+				Object: &buildv1alpha1.Build{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "build.knative.dev/v1alpha1",
+						Kind:       "Build",
+					},
+					Spec: buildv1alpha1.BuildSpec{
+						Steps: []corev1.Container{{
+							Image: "foo",
+						}},
+					},
+				},
+			},
+			RevisionTemplate: RevisionTemplateSpec{
+				Spec: RevisionSpec{
+					Container: corev1.Container{
+						Image: "hellworld",
+					},
+				},
+			},
+		},
+		want: nil,
+	}, {
+		name: "build is missing TypeMeta",
+		c: &ConfigurationSpec{
+			Build: &RawExtension{
+				Object: &buildv1alpha1.Build{
+					Spec: buildv1alpha1.BuildSpec{
+						Steps: []corev1.Container{{
+							Image: "foo",
+						}},
+					},
+				},
+			},
+			RevisionTemplate: RevisionTemplateSpec{
+				Spec: RevisionSpec{
+					Container: corev1.Container{
+						Image: "hellworld",
+					},
+				},
+			},
+		},
+		want: apis.ErrInvalidValue("Object 'Kind' is missing in '{\"metadata\":{\"creationTimestamp\":null},\"spec\":{\"steps\":[{\"name\":\"\",\"image\":\"foo\",\"resources\":{}}],\"timeout\":\"0s\"},\"status\":{\"startTime\":null,\"completionTime\":null,\"stepStates\":null,\"stepsCompleted\":null}}'", "build"),
```

Comment:
Would it be bad to assume, that if there's no TypeMeta then it's Knative Build?

Reply:
I'd rather have this requirement from day 1 of the new shape and it's actually harder to do that with `unstructured.Unstructured`.  Relaxing is easier than making things stricter.